### PR TITLE
Set artifact caching on by default

### LINKED ISSUE #2511: Activate caching by default
At some point when we are confident that it's stable, we should activate artifact caching by default.

### PR DESCRIPTION
**Pull request by priyawadhwa**
Fixes #2511

### CODE REVIEW

File: integration/debug_test.go [hunk 1]
```diff
@@ -40,12 +40,13 @@ func TestDebug(t *testing.T) {
 		{
 			description: "kubectl",
 			dir:         "testdata/debug",
+			args:        []string{"--cache-artifacts=false"},
 			deployments: []string{"jib"},
 			pods:        []string{"nodejs", "npm", "python3"},
 		},
 		{
 			description: "kustomize",
-			args:        []string{"--profile", "kustomize"},
```
Comment (tejal29) [inline integration/debug_test.go]: is there a reason why we turn off caching for these two tests?

Comment (tejal29) [reply integration/debug_test.go]: never mind. just saw your commit message "Add --cache-artifacts=false for integrationt tests where we check builds"

Comment (tejal29) [reply integration/debug_test.go]: but these are debug test? do we still need to disable caching?

**Comment by priyawadhwa** [comment]
cc @nkubala could you PTAL at the event additions and integration test?

**Comment by nkubala** [comment]
eventing and integration test look fine. in the future it might be cool to add some information in the build event itself about whether or not it was built from cache, something like `FromCache` or even `CacheHit`/`CacheMiss`". not sure how useful that would be right now though, just a thought.

**Pull request merged**